lock-api/src/main/java/com/palantir/lock/client/TimelockRpcClient.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client;
+
+import java.util.Set;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+import com.palantir.lock.v2.IdentifiedTimeLockRequest;
+import com.palantir.lock.v2.LockImmutableTimestampResponse;
+import com.palantir.lock.v2.LockRequest;
+import com.palantir.lock.v2.LockResponse;
+import com.palantir.lock.v2.LockToken;
+import com.palantir.lock.v2.StartAtlasDbTransactionResponse;
+import com.palantir.lock.v2.StartIdentifiedAtlasDbTransactionRequest;
+import com.palantir.lock.v2.StartIdentifiedAtlasDbTransactionResponse;
+import com.palantir.lock.v2.WaitForLocksRequest;
+import com.palantir.lock.v2.WaitForLocksResponse;
+import com.palantir.logsafe.Safe;
+import com.palantir.timestamp.TimestampRange;
+
+/**
+ * Interface describing timelock endpoints to be used by feign client factories to create raw clients.
+ *
+ * If you are adding a new endpoint that is a modified version of an existing one, please use the existing naming scheme
+ * in which you add a new version number, rather than describing the additional functionality (i.e. refresh-locks-v2
+ * rather than leasable-refresh-locks)
+ */
+
+@Path("/timelock")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public interface TimelockRpcClient {
+
+    @POST
+    @Path("fresh-timestamp")
+    long getFreshTimestamp();
+
+    @POST
+    @Path("fresh-timestamps")
+    TimestampRange getFreshTimestamps(@Safe @QueryParam("number") int numTimestampsRequested);
+
+    @POST
+    @Path("lock-immutable-timestamp")
+        // TODO (jkong): Can this be deprecated? Are there users outside of Atlas transactions?
```

Comment:
there's some duplicate comments between here and timelockservice; probably should make sure there's a place of truth for them?

Reply:
so like, maybe omit the docs here?